local_pathfinding/objectives.py

```diff
@@ -24,66 +53,313 @@ def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        """Determine the smallest turn angle between s1-s2 and s1-goal
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            ob.Cost: the turning angle from s2 to s1 (degrees)
+        """
+
+        # Calculate the angle of the s1-s2 line segment from North
+        path_direction = math.atan2(s2.getX() - s1.getX(), s2.getY() - s1.getY())
+
+        # Calculate the angle from s1-goal line segment from North
+        global_goal_direction = math.atan2(self.goal_x - s1.getX(), self.goal_y - s1.getY())
+
+        return self.min_turn_angle(global_goal_direction, path_direction)
+
+    def goal_heading_turn_cost(self, s1: ob.SE2StateSpace) -> ob.Cost:
+        """Determine the smallest turn angle between s1-s2 and heading
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            ob.Cost: the turning angle from s2 to s1 (degrees)
+        """
+
+        # Calculate the angle from s1-goal line segment from North
+        global_goal_direction = math.atan2(self.goal_x - s1.getX(), self.goal_y - s1.getY())
+
+        angle = self.min_turn_angle(global_goal_direction, self.heading)
+        return ob.Cost(angle)
+
+    def heading_path_turn_cost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        """Generates the turning cost between s1-s2 and heading of the sailbot
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            ob.Cost: The minimum turning angle between s1-s2 and heading  (degrees)
+        """
+
+        # Calculate the angle of the s1-s2 line segment from North
+        path_direction = math.atan2(s2.getX() - s1.getX(), s2.getY() - s1.getY())
+
+        angle = self.min_turn_angle(path_direction, self.heading)
+        return ob.Cost(angle)
+
+    @staticmethod
+    def min_turn_angle(angle1: float, angle2: float) -> float:
+        """Calculates the minimum turning angle between two angles
+
+        Args:
+            angle1 (float): The first angle in radians
+            angle2 (float): The second angle in radians
+
+        Returns:
+            float: The minimum turning angle between the two angles (degrees)
+        """
+
+        # Calculate the uncorrected turn size [0, 2pi]
+        turn_size_bias = math.fabs(angle1 - angle2)
+
+        # Correct the angle in between [0, pi]
+        if turn_size_bias > math.pi:
+            turn_size_unbias = 2 * math.pi - turn_size_bias
+        else:
+            turn_size_unbias = turn_size_bias
+
+        return math.degrees(math.fabs(turn_size_unbias))
```

Comment:
I could be wrong but it seems like the `fabs` function isn't needed in this line? In the `if` block, `turn_size_unbias` is $2 \pi - \text{ angle between } (\pi, 2\pi]$, and in the else block, `turn_size_unbias` is already positive due to the `fabs` on line 246.